### PR TITLE
318 bug activate users

### PR DESCRIPTION
**Pull request by reabiliti**
https://github.com/howtohireme/give-me-poc/issues/318

### Description

Update pundit to 2.0.0.beta
Fix rights to activate candidates

### Testing steps

* Current user has role staff
* in user management sees a button Review applicant in the candidate with the status screening completed
* in Applicants can only activate users with the status screening completed
* Current user has role mentor
* in user management sees a button Review applicant in the candidate with the status screening completed if primary skill equal.
* in Applicants can only activate users with the status screening completed and equal primary skill

### Features PR URL

https://github.com/howtohireme/give-me-poc-features/pull/19

### Checklist

Make sure that all steps a checked before the merge

- [x] RSpec tests are passing on CI
- [x] Cucumber features are passing localy
- [x] `bin/cop -a` does not return any warnings
- [x] Tested manually

### CODE REVIEW

File: app/policies/test_task_assignment_policy.rb [hunk 1]
```diff
@@ -7,19 +7,19 @@ def index?
   end
 
   def show?
-    staff_or_mentor?
+    staff_assignment? || mentor_assignment?
```
Comment (khusnetdinov) [inline app/policies/test_task_assignment_policy.rb]: Если одинакое то нужно вынести в метод и другие методы делегировать на этот метод

File: spec/policies/web/dashboard/idea_policy_spec.rb [hunk 2]
```diff
@@ -55,3 +55,4 @@
     it { is_expected.not_to permit_action(:reject) }
   end
 end
+# TODO: write tests for scope as test task assignment
```
Comment (khusnetdinov) [inline spec/policies/web/dashboard/idea_policy_spec.rb]: Не оставлять TODO, доделыать

Comment (Mehonoshin) [reply spec/policies/web/dashboard/idea_policy_spec.rb]: либо создавать отдельный тикет, и ссылку на него в TODO оставлять

Comment (Mehonoshin) [reply spec/policies/web/dashboard/idea_policy_spec.rb]: Вообще у нас бот должен быть для этого))

**Comment by reabiliti** [comment]
fix features

File: app/controllers/web/dashboard/test_task_assignments_controller.rb [hunk 3]
```diff
@@ -5,7 +5,7 @@ module Dashboard
     # Handles all management logic for newcomers
     class TestTaskAssignmentsController < BaseController
       before_action :candidate, only: %i[show activate reject]
-      before_action :authorize_staff!
+      before_action :authorize_staff!, only: :index
```
[the remark this answers is not in the document]
Comment (reabiliti) [reply app/controllers/web/dashboard/test_task_assignments_controller.rb]: Мы теперь для остальны действий проверяем этим методом права:
      ```def candidate
        @candidate ||= authorize User.find(params[:id]), policy_class: TestTaskAssignmentPolicy
      end```

**Pull request merged**